driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -1339,42 +1339,120 @@ static __always_inline void apply_dynamic_snaplen(struct pt_regs *regs, uint16_t
 	 *  - recvmsg
 	 *  - sendmsg
 	 */
-	unsigned long args[3];
-	extract__network_args(args, 3, regs);
 
-	/* All the syscalls involved in this logic have the `fd` as first syscall argument */
-	int32_t socket_fd = (int32_t)args[0];
-	if(socket_fd < 0)
-	{
-		return;
-	}
+	int32_t socket_fd = 0;
+	uint16_t port_local = 0;
+	uint16_t port_remote = 0;
+	uint16_t socket_family = 0;
+	unsigned long args[6];
+	bool extracted = true;
+	struct sockaddr *sockaddr;
 
-	struct file *file = extract__file_struct_from_fd(socket_fd);
-	struct socket *socket = BPF_CORE_READ(file, private_data);
-	struct sock *sk = BPF_CORE_READ(socket, sk);
-	if(!sk)
+	switch(syscall_number)
 	{
-		return;
-	}
+		case SCN_SENDTO:
+		{
+			extract__network_args(args, 6, regs);
```

Comment:
I am wondering whether we could just pass the `socket_fd_arg_num` here, and that's it, instead of the `enum`?

Reply:
The 2 syscall are slightly different and I can't apply the same logic to both syscall 😅